astropy/units/tests/test_units.py

```diff
@@ -329,23 +329,14 @@ def test_empty_compose():
         u.m.compose(units=[])
 
 
-def _unit_as_str(unit):
-    # This function serves two purposes - it is used to sort the units to
-    # test alphabetically, and it is also use to allow pytest to show the unit
-    # in the [] when running the parametrized tests.
-    return str(unit)
-
-
 # We use a set to make sure we don't have any duplicates.
 COMPOSE_ROUNDTRIP = set()
 for val in u.__dict__.values():
     if isinstance(val, u.UnitBase) and not isinstance(val, u.PrefixUnit):
         COMPOSE_ROUNDTRIP.add(val)
 
 
-@pytest.mark.parametrize(
-    "unit", sorted(COMPOSE_ROUNDTRIP, key=_unit_as_str), ids=_unit_as_str
-)
+@pytest.mark.parametrize("unit", sorted(COMPOSE_ROUNDTRIP, key=str), ids=repr)
```

Comment:
I'm using `repr()` here for the `ids` because `str(u.dimensionless_unscaled)` returns an empty string.